src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Models/AsAzureClientSession.cs

```diff
@@ -136,14 +119,20 @@ public AsAzureProfile Login(AsAzureContext asAzureContext)
 
         public static string GetAuthorityUrlForEnvironment(AsAzureEnvironment environment)
         {
-            var environmentKey = AsAzureRolloutEnvironmentMapping.Keys.FirstOrDefault(s => environment.Name.Contains(s));
-            AsAzureAuthInfo authInfo = null;
-            if (string.IsNullOrEmpty(environmentKey) || !AsAzureRolloutEnvironmentMapping.TryGetValue(environmentKey, out authInfo))
+            var profile = AzureRmProfileProvider.Instance.GetProfile<AzureRmProfile>();
+            var availableEnvironments = profile.Environments;
```

Comment:
All you should need to do here is get DefaultContext.Environment - this will have the environment the user is currently logged in to.

Reply:
Do you mean DefaultContext in AzureRMCmdlet?

Our data plane cmdlets inherit from AzurePSCmdlet and do not depend on Login-AzureRmAccount at all. We have Add-AzureAnalysisServicesAccount to login to a specific rollout (region), not environment (cloud).

If we instead depends on the state of what users do with Login-AzureRmAccount, it's a protocol change to our customers. It will be a really big change and probably needs PM involved.

Please correct me if I misunderstand something.

Reply:
>             All you should need to do here is get DefaultContext.Environment - this will have the environment the user is currently logged in to.

This is breaking, because before this, customers who are using these dataplane cmdlets did not have to login to Azure RM. Now we are requiring them to login to RM to get  dataplane related data. @markcowl , is there a way to get this information without having to login. Because in our scenarios, the user who is logging to Azure ASAccount is a server admin (internal to ASAZure) and has nothing to do with the Azure Resource. The user will not have any access to subscription, so Add-AzureRM will fail. So this is actually more than just a breaking change, basically disabling the usecase. :)

Reply:
The issue is that you should not need to use Authentication.ResourceManager dll - I believe the only reason you are using it is to get at the environment, but you should be able to do this with:

```
AzureRmProfileProvider.Instance.Profile.DefaultContext.Environment
```

this returns the current environment withotu having to take any dependencies on AzureRMProfile at all, and with this, you should be able to remove this library dependency

Reply:
@markcowl , If the user has not logged in to RM, what value will be returned by AzureRmProfileProvider.Instance.Profile.DefaultContext.Environment?

Reply:
Well, then, it seems like your current code won't work either, so I'm not sure what you are asking.  This is precisely the same as using AzureRMProfileProivder.Instance.GetProfile(), which is what you are currently doing, except that it doesn't require the additional assembly dependency.  So if this won't work, your proposed code also won't work

Reply:
Yes that I totally agree. @pychuang do we need this PR at all then?

Reply:
Agree. This PR won't work either. We probably need to go back to the original approach.

I will create a PR with original approach, and we can continue the discussion then.